### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Leidse**

### CODE REVIEW

File: src/filter.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    int i = 1;	
+    //TODO: Change to use for loop instead of while
+    while ( i < length ) {
+		for (int j = i; list[j - 1] > list[j]; j-- ) {
+			int temp = list[j];
+			list[j] = list[j - 1];
+			list[j - 1] = temp;
+		}
+		i++;
+	}
```
Comment (elsavira) [inline src/filter.cpp]: Wrong indentation here. Use clang-Format

File: src/lidar_mini.hpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * @license   MIT
+ */
+
+/**
+ * @class LIDARmini
+ * @brief Class for the TF LIDAR Mini
+ */
+class LIDARmini {
+private:
+    hwlib::target::pin_in RX;
```
Comment (elsavira) [inline src/lidar_mini.hpp]: Explain what RX is, or change the name.

File: src/lidar_mini.hpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * @author    Wouter Dijksta and Kiet van Osnabrugge
+ * @date      21-5-2018
+ * @license   MIT
+ */
+
+/**
+ * @class LIDARmini
+ * @brief Class for the TF LIDAR Mini
+ */
+class LIDARmini {
```
Comment (elsavira) [inline src/lidar_mini.hpp:19]: Indents incorrect in this class. Use clang-format as defined on the wiki.

File: src/filter.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#include "filter.hpp"
+
+/**
+ * @file      filter.cpp
+ * @brief     Cpp file for the class filter which will contain differnet filters
```
Comment (elsavira) [inline src/filter.cpp]: different*

File: src/filter.cpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * @license   MIT
+ */
+
+//TODO: Change to use C++ array
+int Filter::getMedian(int list[], int length)
+{    
+    int i = 1;	
+    //TODO: Change to use for loop instead of while
+    while ( i < length ) {
+		for (int j = i; list[j - 1] > list[j]; j-- ) {
```
Comment (elsavira) [inline src/filter.cpp]: Tabs instead of spaces. Use 4 spaces for indentation according to code guidelines.

File: src/uart_protocol.cpp [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * @file      uart_protocol.cpp
+ * @brief     Cpp file for the class UARTProtocol which will contain functions use a UARTProtocol which is needed for the LIDARmini class.
+ * @author    Wouter Dijksta and Kiet van Osnabrugge
+ * @date      21-5-2018
+ * @license   MIT
+ */
+
+enum uartEnum {
```
Comment (elsavira) [inline src/uart_protocol.cpp]: use enum class instead. C++ style.

File: src/uart_protocol.cpp [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    sizeDataPackage = 8
+};
+
+UARTProtocol::UARTProtocol(hwlib::target::pin_in& RX, int waitStartByteCycles):
+    RX(RX),
+    waitStartByteCycles(waitStartByteCycles)
+{}
+
+
+char UARTProtocol::getByte()
```
Comment (elsavira) [inline src/uart_protocol.cpp]: Brackets not according to clang-format guidelines. See wiki.

File: src/uart_protocol.hpp [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+        UARTProtocol(hwlib::target::pin_in& RX, int waitStartByteCycles);
+
+        /**
+    	* @brief gets a byte form the RX pin
+        *
+        * This function gets a byte one bit at a time from the RX pin.
+        *
+        * @return 
+        * char : returns a char (which corrosponds to a byte)
```
Comment (elsavira) [inline src/uart_protocol.hpp]: corresponds*

**Review by elsavira — CHANGES_REQUESTED** [review]
Make sure clang-format is working properly in your IDE. 
Explain variable names where necessary.

**Comment by elsavira** [comment]
Also update the travis tests with the new template files before re-requesting.

**Review by elsavira — CHANGES_REQUESTED** [review]
Implement the new travis tests into the module. (Pull the Build-environment, copy all test files from the template-arduino folder, and replace the ones in your current module folder.) 

Sorry about the RX pin, didn't know.

After fixing that, the request is accepted.

**Comment by elsavira** [comment]
I can no longer accept this pull request for Sprint 2, since features from Sprint 3 have been pushed to the dev. Refrain from pushing new features to the dev until old features have been succesfully pushed to master. Please recreate this pull request for the new sprint.

File: src/lidar_mini.hpp (hunk 3 above)
Comment (Leidse) [reply src/lidar_mini.hpp:19]: I have used the clang-format plugin and dont see what is wrong

File: src/lidar_mini.hpp (hunk 2 above)
Comment (Leidse) [reply src/lidar_mini.hpp]: RX is a pin on the arduino and pretty self explanertory

**Comment by elsavira** [comment]
Please make a NEW pull request with the correct title for the new slice.

**Pull request closed without merging**